kubernetes/cache/kube_cache.go

```diff
@@ -1464,11 +1464,21 @@ func (c *kubeCache) GetTelemetries(namespace, labelSelector string) ([]*v1alpha1
 	return retTelemetries, nil
 }
 
+func (c *kubeCache) IsK8sGatewayListerInit(namespace string) bool {
+	// Read lock will prevent the cache from being refreshed while we are reading from the lister
+	// but it won't prevent other routines from reading from the lister.
+	defer c.cacheLock.RUnlock()
+	c.cacheLock.RLock()
+	return c.getCacheLister(namespace).k8sgatewayLister != nil
+}
+
 func (c *kubeCache) GetK8sGateway(namespace, name string) (*gatewayapi_v1.Gateway, error) {
 	if err := checkIstioAPIsExist(c.client); err != nil {
 		return nil, err
 	}
-
+	if !c.IsK8sGatewayListerInit(namespace) {
+		return nil, nil
```

Comment:
I think we should always return an error here instead of `nil, nil`. Whatever is calling `GetK8sGateway` can handle the case where the CRDs are installed after the server is started and either log a warning/error or send an error response to the user telling them to restart their kiali server to work with the new CRDs.

It also seems like we should just check this once at startup. Maybe that info can be saved on the client. Not necessary to change for this fix.

Reply:
Afaik when CRDs are installed, it is not mandatory that the cluster is configured to use GW API, so showing user an error/warning message in UI can be confusing, so better to log a message. Also as it is not an error/warning, it can be an info message, as we have in documentations that Kiali needs to be restarted.
The issue happens after the Kiali is started, when CRDa are created, we cannot detect it while starting.
I thought that would be odd to add one more variable in `kubeCache` describing whether listers are inited or not.

Reply:
This method still needs to return an error when it is returning a `nil` value. Returning `nil, nil` will lead to runtime panics for anything that calls this and only checks `err != nil` before trying to use the return value. Actually taking a second look these methods aren't being used anywhere right now which I think is why there is no runtime error right now but if they were used they'd be used here: https://github.com/kiali/kiali/blob/95e4b4975a81ae486aa2badefc102237cd737b30/business/istio_config.go#L539-L543

Since just the `err` is checked before accessing `istioConfigDetail.K8sGateway.Kind`, which is a perfectly reasonable thing to do, you'd be trying to access a field on a nil pointer and that'd be a runtime error.